app/controllers/application_controller.rb

```diff
@@ -3,4 +3,15 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
   include SessionsHelper
+
+  private
+
+  def query_header_for_query(query, any_search_results)
```

Comment:
I moved this from items_controller into application_controller so that it would be available in all controllers.  I guess we could make it a helper, but to use it in a view, we would need to capture the query in an instance variable.  Is that better?

I've never used concerns, so I don't know what we gain with them... :-/